guarddog/analyzer/metadata/bundled_binary.py

```diff
@@ -28,21 +29,38 @@ def detect(
         name: Optional[str] = None,
         version: Optional[str] = None,
     ) -> tuple[bool, str]:
+        def format_file(file: str, kind: str) -> str:
+            return f"{file} ({kind})"
+
+        def sha256(file: str) -> str:
+            with open(file, "rb") as f:
+                hasher = hashlib.sha256()
+                while (chunk := f.read(4096)):
+                    hasher.update(chunk)
+                return hasher.hexdigest()
 
         log.debug(
             f"Running bundled binary heuristic on package {name} version {version}"
         )
         if not path:
             raise ValueError("path is needed to run heuristic " + self.get_name())
 
-        bin_files = []
+        bin_files = {}
         for root, _, files in os.walk(path):
             for f in files:
-                kind = self.is_binary(os.path.join(root, f))
+                path = os.path.join(root, f)
+                kind = self.is_binary(path)
                 if kind:
-                    bin_files.append(f"{f} type {kind}")
+                    digest = sha256(path)
+                    if digest not in bin_files:
+                        bin_files[digest] = [format_file(f, kind)]
+                    else:
+                        bin_files[digest].append(format_file(f, kind))
         if bin_files:
-            return True, "Binary file/s detected in package: " + reduce(lambda x, y: f"{x}, {y}", bin_files)
+            output_lines = '\n'.join(
```

Comment:
For readability, I would invert the condition to 
```
if not bin_files: 
    return False, ""

output_lines...
```

Reply:
Added in [e7020d0](https://github.com/DataDog/guarddog/pull/471/commits/e7020d0d0e49c990a7b061a347a85605afc10620)